spec/classes/foreman_proxy__tftp__netboot_spec.rb

```diff
@@ -50,6 +52,18 @@
             it { should contain_file("/tftproot/grub2/grubx64.efi").with_source('/boot/efi/EFI/centos/grubx64.efi') }
             it { should contain_file("/tftproot/grub2/shim.efi").with_source('/boot/efi/EFI/centos/shim.efi').with_owner('root').with_mode('0644') }
           end
+        elsif facts[:operatingsystemmajrelease].to_i == 8
+          case facts[:operatingsystem]
+          when /^(RedHat|Scientific|OracleLinux)$/
+            it { should contain_file("/tftproot/grub2/grubx64.efi").with_source('/boot/efi/EFI/redhat/grubx64.efi') }
+            it { should contain_file("/tftproot/grub2/shimx64.efi").with_source('/boot/efi/EFI/redhat/shimx64.efi').with_owner('root').with_mode('0644') }
+          when 'Fedora'
+            it { should contain_file("/tftproot/grub2/grubx64.efi").with_source('/boot/efi/EFI/fedora/grubx64.efi') }
+            it { should contain_file("/tftproot/grub2/shimx64.efi").with_source('/boot/efi/EFI/fedora/shimx64.efi').with_owner('root').with_mode('0644') }
+          when 'CentOS'
+            it { should contain_file("/tftproot/grub2/grubx64.efi").with_source('/boot/efi/EFI/centos/grubx64.efi') }
+            it { should contain_file("/tftproot/grub2/shimx64.efi").with_source('/boot/efi/EFI/centos/shimx64.efi').with_owner('root').with_mode('0644') }
+          end
         else
           it { is_expected.to contain_class('foreman_proxy::tftp::netboot').with_grub_installation_type('redhat_old') }
```

Comment:
Perhaps this is a good time to drop the EL6 code since this module doesn't support EL6. We also don't support running on EL 7.4 so all of that can be cleaned up.

Reply:
I will have to research this further. I assume it involves changes to multiple manifests and tests.

Reply:
https://github.com/theforeman/puppet-foreman_proxy/pull/590